packages/dev/core/src/Maths/math.vector.functions.ts

```diff
@@ -29,3 +32,100 @@ export function Vector3ToFixed(vector: Vector3, decimalCount: number): string {
 export function Vector4ToFixed(vector: Vector4, decimalCount: number): string {
     return `{X: ${vector.x.toFixed(decimalCount)} Y: ${vector.y.toFixed(decimalCount)} Z: ${vector.z.toFixed(decimalCount)} W: ${vector.w.toFixed(decimalCount)}}`;
 }
+
+/**
+ * Creates a new Vector2 from the given Vector3, ommiting the z coordinate.
+ * @returns a new Vector2 set from the given Vector3, ommiting the z coordinate.
+ */
+export function Vector2FromVector3(vector: Vector3): Vector2 {
+    return new Vector2(vector._x, vector._y);
+}
+
+/**
+ * Creates a new Vector3 that gets all its element initialized by the same scalar.
+ * @param scalar 
+ * @returns a new Vector3 from the given scalar.
+ */
+export function Vector3FromScalar(scalar: number): Vector3 {
+    return new Vector3(scalar, scalar, scalar);
+}
+/**
+ * Creates a new translation matrix that contains a translation by the specified vector.
+ * @param vector The vector that contains a translation.
+ * @returns a new matrix that translates by the specified vector.
+ */
+export function TranslationMatrixFromVector3(vector: Vector3): Matrix {
+    return Matrix.Translation(vector._x, vector._y, vector._z);
+}
+
+/**
+ * Creates a new translation matrix that contains a translation by the specified Vector2.
+ * The z coordinate is initialized with 0.
+ * @param vector 
+ * @returns a new matrix containing a translations by vector.x, vector.y, 0.
+ */
+export function TranslationMatrixFromVector2(vector: Vector2): Matrix {
+    return Matrix.Translation(vector.x, vector.y, 0);
+}
+
+/**
+ * Creates a new matrix that scales all dimensions by the specified scalar.
+ * @param scalar 
+ * @returns a new scaling matrix that contains a scaling by the specified scalar.
+ */
+export function ScalingMatrixFromScalar(scalar: number): Matrix {
+    return Matrix.Scaling(scalar, scalar, scalar);
+}
+
+/**
+ * Writes a matrix composed by merging scale (vector3), rotation (quaternion) and translation (vector3)
+ * directly to a FloatArray at the given offset.
+ * Example Playground - https://playground.babylonjs.com/#AV9X17#25
+ * @param scale defines the scale vector3
+ * @param rotation defines the rotation quaternion
+ * @param translation defines the translation vector3
+ * @param array defines the array to write to
+ * @param offset defines the offset in the target array where to start storing values
+ */
+export function ComposeMatrixToArray<T extends Matrix>(scale: DeepImmutable<Vector3>, rotation: DeepImmutable<Quaternion>, translation: DeepImmutable<Vector3>, array: FloatArray, offset: number): void {
+    const x = rotation._x,

```

Comment:
Isn't it a code duplication of what we already have?

Reply:
Yes as mentioned in the forum, this is code duplication of `Matrix.ComposeToRef`, unfortunately I don't know how to reuse the existing code. Maybe `ComposeToRef` could make use of this version, but then you would have the `+ offset` with an offset of 0.